modules/services/ssh-agent.nix

```
@@ -0,0 +1,36 @@
+{ config, options, lib, pkgs, ... }:
+
+let
+
+  cfg = config.services.ssh-agent;
+
+in {
+  meta.maintainers = [ lib.maintainers.lheckemann ];
+
+  options = {
+    services.ssh-agent = {
+      enable = lib.mkEnableOption "OpenSSH private key agent";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.sessionVariablesExtra = ''
+      if [[ -z "$SSH_AUTH_SOCK" ]]; then
+        export SSH_AUTH_SOCK=$XDG_RUNTIME_DIR/ssh-agent.sock
```

We should probably use the same path that is used by nixpkgs: https://github.com/NixOS/nixpkgs/blob/c57bc5f8129dd25337ad155f22fed3e3809dfe67/nixos/modules/programs/ssh.nix#L347-L352. That way, if people have `programs.ssh.startAgent` in their nixos config, it won't conflict with this option.

```suggestion
        export SSH_AUTH_SOCK=$XDG_RUNTIME_DIR/ssh-agent
```